Packs/CyberArkEPM/Integrations/CyberArkEPMEventCollector/CyberArkEPMEventCollector.py

```diff
@@ -0,0 +1,470 @@
+def get_policy_audits_command(client: Client, last_run: dict, limit: int | None) -> tuple[list, CommandResults]:
+    """ Fetches policy audits from CyberArkEPM
+    Args:
+        client (Client): CyberArkEPM client to use.
+        last_run (dict): The last run {set_id: {from_date: 01-01-2023T00:00:00.123Z, next_cursor: 123456}}.
+        limit (int): The maximum policy audits to return per set_id default is 5.
+    Return:
+        (list, CommandResults) A list of policy audits to push to XSIAM, A CommandResults object.
+    """
+    results = get_policy_audits(client, last_run, limit)  # type: ignore
+    events_list_of_lists = [value.get('events', []) for value in results.values()]
+    events_list = list(chain(*events_list_of_lists))
+    human_readable = tableToMarkdown('Policy Audits', events_list)
+
+    return events_list, CommandResults(readable_output=human_readable, raw_response=events_list)
+
+
+def get_detailed_events_command(client: Client, last_run: dict, limit: int | None = 5) -> tuple[list, CommandResults]:
+    """ Fetches evens from CyberArkEPM
+    Args:
+        client (Client): CyberArkEPM client to use.
+        last_run (dict): The last run {set_id: {from_date: 01-01-2023T00:00:00.123Z, next_cursor: 123456}}.
+        limit (int): The maximum events to return per set_id default is 5.
+    Return:
+        (list, CommandResults) A list of events to push to XSIAM, A CommandResults object.
+    """
+    results = get_detailed_events(client, last_run, limit)  # type: ignore
+    events_list_of_lists = [value.get('events', []) for value in results.values()]
+    events_list = list(chain(*events_list_of_lists))
+    human_readable = tableToMarkdown('Events', events_list)
+
+    return events_list, CommandResults(readable_output=human_readable, raw_response=events_list)
+
+
+def fetch_events(client: Client, last_run: dict, max_fetch: int = MAX_FETCH) -> tuple[list, dict]:
+    """ Fetches 3 types of events from CyberArkEPM
+        - admin_audits
+        - policy_audits
+        - events
+    Args:
+        client (Client): CyberArkEPM client to use.
+        last_run (dict): The last run information.
+        max_fetch (int): The max events to return per fetch default is 250.
+    Return:
+        (list, dict) A list of events to push to XSIAM, A dict with information for next fetch.
+    """
+    events: list = []
+    demisto.info(f'Start fetching last run: {last_run}')
+
+    for set_id, admin_audits in get_admin_audits(client, last_run, max_fetch).items():
+        if admin_audits:
+            last_run[set_id]['admin_audits']['from_date'] = prepare_datetime(admin_audits[-1].get('EventTime'), increase=True)
+            events.extend(admin_audits)
+
+    for set_id, policy_audits_last_run in get_policy_audits(client, last_run, max_fetch).items():
+        if policy_audits := policy_audits_last_run.get('events', []):
+            last_run[set_id]['policy_audits']['next_cursor'] = policy_audits_last_run.get('next_cursor')
```

Comment:
block from line 357  and block from 366 are the same. consider extracting to a single method to prevent code dup